protocol/lavaprotocol/finalization_consensus.go

```diff
@@ -188,62 +217,51 @@ func (fc *FinalizationConsensus) NewEpoch(epoch uint64) {
 			utils.LavaFormatDebug("finalization information epoch changed", utils.Attribute{Key: "specId", Value: fc.specId}, utils.Attribute{Key: "epoch", Value: epoch})
 		}
 		// means it's time to refresh the epoch
-		fc.prevEpochProviderHashesConsensus = fc.currentProviderHashesConsensus
-		fc.currentProviderHashesConsensus = []ProviderHashesConsensus{}
+		fc.prevEpochBlockToHashesToAgreeingProviders = fc.currentBlockToHashesToAgreeingProviders
+		fc.currentBlockToHashesToAgreeingProviders = BlockToHashesToAgreeingProviders{}
 		fc.currentEpoch = epoch
 	}
 }
 
-func (fc *FinalizationConsensus) LatestBlock() uint64 {
-	fc.providerDataContainersMu.RLock()
-	defer fc.providerDataContainersMu.RUnlock()
-	return fc.latestBlockByMedian
-}
-
-func (fc *FinalizationConsensus) GetExpectedBlockHeights(averageBlockTime_ms time.Duration) map[string]int64 {
-	var highestBlockNumber int64 = 0
-	FindAndUpdateHighestBlockNumber := func(listProviderHashesConsensus []ProviderHashesConsensus) {
-		for _, providerHashesConsensus := range listProviderHashesConsensus {
-			for _, providerDataContainer := range providerHashesConsensus.agreeingProviders {
-				if highestBlockNumber < providerDataContainer.LatestFinalizedBlock {
-					highestBlockNumber = providerDataContainer.LatestFinalizedBlock
-				}
-			}
-		}
-	}
-
-	FindAndUpdateHighestBlockNumber(fc.prevEpochProviderHashesConsensus) // update the highest in place
-	FindAndUpdateHighestBlockNumber(fc.currentProviderHashesConsensus)
+func (fc *FinalizationConsensus) getExpectedBlockHeightsOfProviders(averageBlockTime_ms time.Duration) map[string]int64 {
```

Comment:
add a nice comment above explaining what the method is doing. complicated methods deserve a nice explanation.

Reply:
Added a nice comment in 36df2d638a6373615eeaeb8bda88e58fd7d03412